software/script/chameleon_cli_unit.py

```diff
@@ -1022,6 +1023,32 @@ def res_value(self, src_blk, src_type, src_key, dst_blk, dst_type, dst_key):
             print(f" - {CR}Restore fail.{C0}")
 
 
+_KEY = re.compile("[a-fA-F0-9]{12}", flags=re.MULTILINE)
+
+
+def _run_mfkey32v2(items):
+    output_str = subprocess.run(
+        [
+            "mfkey32v2.exe" if sys.platform == "win32" else "./mfkey32v2",
+            items[0]["uid"],
+            items[0]["nt"],
+            items[0]["nr"],
+            items[0]["ar"],
+            items[1]["nt"],
+            items[1]["nr"],
+            items[1]["ar"],
+        ],
```

Comment:
Personally needed to add `shell=True` in the `subprocess.run()` args
Running this one a windows machine, not sure how this will affect a different OS
```suggestion
        ],
        shell=True,
```

Reply:
Thanks! Can you confirm that with this patch you needed to add `shell=True`? With the exact same command, without changing anything else?
That's very surprising to me! Could you share the output without adding `shell=True`?

Reply:
Sure! Here it is without `shell=True`

```
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/mingw64/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/home/USER/ChameleonUltra/software/script/chameleon_cli_unit.py", line 1030, in _run_mfkey32v2
    output_str = subprocess.run(
                 ^^^^^^^^^^^^^^^
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/mingw64/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/mingw64/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/mingw64/lib/python3.11/subprocess.py", line 1538, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 2] The system cannot find the file specified
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/home/USER/ChameleonUltra/software/script/chameleon_cli_main.py", line 144, in exec_cmd
    raise error
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/home/USER/ChameleonUltra/software/script/chameleon_cli_main.py", line 139, in exec_cmd
    unit.on_exec(args_parse_result)
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/home/USER/ChameleonUltra/software/script/chameleon_cli_unit.py", line 1137, in on_exec
    result_maps[uid][block]['A'] = self.decrypt_by_list(records)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/home/USER/ChameleonUltra/software/script/chameleon_cli_unit.py", line 1076, in decrypt_by_list
    for key in pool.imap(
  File "C:/Users/USER/Desktop/Chamilion Ultra/CLI/ProxSpace/msys2/mingw64/lib/python3.11/multiprocessing/pool.py", line 873, in next
    raise value
FileNotFoundError: [WinError 2] The system cannot find the file specified

```

Reply:
Oh OK, I'm having a look. Thanks for your answer!

Reply:
Can you try if it if fixed @MrFlou?